app/graphql/mutations/goal/delete_goal.rb

```diff
@@ -5,15 +5,15 @@ class Goal::DeleteGoal < BaseMutation
     argument :goal_id, ID, required: true, description: 'The ID of the goal to delete'
 
     field :goal_id, ID, null: true
-    field :errors, [String], null: false
 
     def resolve(goal_id:)
       goal = ::Goal.find(goal_id)
 
       if goal.destroy
-        { goal_id: goal.id, errors: [] }
+        { goal_id: goal.id }
       else
-        { goal_id: nil, errors: goal.errors.full_messages }
+        Util::ErrorBuilder.build_errors(context, goal.errors)
```

Comment:
I'm curious why you switched from raising the error to adding the errors to the context? Raising is a lot clearer IMO.

Reply:
I couldn't find a way to raise a GraphQL error with multiple fields without adding them to the context.